datafusion/substrait/src/logical_plan/consumer.rs

```diff
@@ -1342,3 +1285,67 @@ fn from_substrait_null(null_type: &Type) -> Result<ScalarValue> {
         ))
     }
 }
+
+async fn make_datafusion_like(
+    case_insensitive: bool,
+    f: &ScalarFunction,
+    input_schema: &DFSchema,
+    extensions: &HashMap<u32, &String>,
+) -> Result<Arc<Expr>> {
+    let fn_name = if case_insensitive { "ILIKE" } else { "LIKE" };
+    if f.arguments.len() != 3 {
+        return Err(DataFusionError::NotImplemented(format!(
+            "Expect three arguments for `{fn_name}` expr"
+        )));
+    }
+
+    let Some(ArgType::Value(expr_substrait)) = &f.arguments[0].arg_type else {
+        return Err(DataFusionError::NotImplemented(
+            format!("Invalid arguments type for `{}` expr", fn_name)
+        ))
+    };
```

Comment:
I'm not sure if
```Rust
format!("Invalid arguments type for `{}` expr", fn_name)
```
or 
```Rust
format!("Invalid arguments type for `{fn_name}` expr")
```
is preferred in Rust, but maybe we should choose one to be consistent here? It may be confusing to the reader if different syntaxes are used to implement the same semantic.

Reply:
Okay 👍 

BTW, clippy has a lint about this style [`uninlined_format_args`](https://rust-lang.github.io/rust-clippy/master/index.html#/uninlined_format_args). (but maybe it's unnecessary to enable it?

Reply:
> BTW, clippy has a lint about this style [uninlined_format_args](https://rust-lang.github.io/rust-clippy/master/index.html#/uninlined_format_args). (but maybe it's unnecessary to enable it?

Oh thanks for the reference! Inlining the args make sense.